parsers/IN_MH.py

```diff
@@ -145,20 +148,37 @@ def fetch_production(
     }
 
     image = imread(url)
-    image = Image.fromarray(image)  # create PIL image
+    # In certain scenario, the URL returns a blank image. Let's verify if the image was read is of proper size
+    if image.size == 0:
+        raise ParserException(
+            "IN_MH.py",
+            "Invalid data read from the source, the source might not be available",
+        )
 
-    imgs = [read(loc["value"], image) for loc in locations.values()]
+    image = Image.fromarray(image)  # create PIL image
 
-    # string together all image sections and recognize resulting line
-    imgs_line = np.hstack(list(np.asarray(i) for i in imgs[:]))
-    imgs_line = Image.fromarray(imgs_line)
-    text = pytesseract.image_to_string(imgs_line, lang="digits_comma", config="--psm 7")
-    text = text.split(" ")
+    # Read small images for each loaction's bounding box from the main image
+    imgs = [read_image_sections(loc["value"], image) for loc in locations.values()]
 
-    # generate dict from string list
     values = {}
-    for count, key in enumerate(locations):
-        values[key] = max([float(text[count]), 0])
+
+    # for each location, convert the image to a float integer and add it in the map corresponding to the key
+    for index, key in enumerate(locations):
+        digit_text = pytesseract.image_to_string(
+            imgs[index], lang="digits_comma", config="--psm 7"
+        )
+        val = 0
+        try:
+            val = float(digit_text)
+        except ValueError:
+            # If the image cannot be converted to a valid number, log an error but do not break the parser
+            val = 0
+            logger.error(
```

Comment:
> Making the fix future-proof and fault tolerant to such misalignment of coordinates of the values
In the future, I am assuming that the coordinates of the boxes may go wrong, maybe because the website will change the layout of the image, that's an assumption.
This change in the algorithm will read fields as much as possible and skip those that it will not be able to read for some reason and log an error. It will not misinterpret any legit value, but instead, skip over fields pytesseract is not able to interpret.

While this would make the parser more fault tolerant it might also have a negative effect of returning partial data. Just making a note of it so others are aware.